e2e/cypress/support/ui_commands.js

```diff
@@ -216,6 +216,106 @@ Cypress.Commands.add('compareLastPostHTMLContentFromFile', (file, timeout = TIME
     });
 });
 
+// ***********************************************************
+// DM
+// ***********************************************************
+
+/**
+ * Sends a DM to a given user
+ * @param {User} user - the user that should get the message
+ * @param {String} message - the message to send
+ */
+Cypress.Commands.add('sendDirectMessageToUser', (user, message) => {
+    // # Open a new direct message with firstDMUser
+    cy.get('#addDirectChannel').click();
+
+    // # Type username
+    cy.get('#selectItems input').should('be.enabled').type(`@${user.username}`, {force: true});
+
+    // * Expect user count in the list to be 1
+    cy.get('#multiSelectList').
+        should('be.visible').
+        children().
+        should('have.length', 1);
+
+    // # Select first user in the list
+    cy.get('body').
+        type('{downArrow}').
+        type('{enter}');
+
+    // # Click on "Go" in the group message's dialog to begin the conversation
+    cy.get('#saveItems').click();
+
+    // * Expect the channel title to be the user's username
+    // In the channel header, it seems there is a space after the username, justifying the use of contains.text instead of have.text
+    cy.get('#channelHeaderTitle').should('be.visible').and('contain.text', user.username);
+
+    // # Type message and send it to the user
+    cy.get('#post_textbox').
+        type(message).
+        type('{enter}');
+});
+
+/**
+ * Sends a GM to a given user list
+ * @param {User[]} users - the users that should get the message
+ * @param {String} message - the message to send
+ */
+Cypress.Commands.add('sendDirectMessageToUsers', (users, message) => {
+    // # Open a new direct message
+    cy.get('#addDirectChannel').click();
+
+    users.forEach((user) => {
+        // # Type username
+        cy.get('#selectItems input').should('be.enabled').type(`@${user.username}`, {force: true});
+
+        // * Expect user count in the list to be 1
+        cy.get('#multiSelectList').
+            should('be.visible').
+            children().
+            should('have.length', 1);
+
+        // # Select first user in the list
+        cy.get('body').
+            type('{downArrow}').
+            type('{enter}');
+    });
+
+    // # Click on "Go" in the group message's dialog to begin the conversation
+    cy.get('#saveItems').click();
+
+    // * Expect the channel title to be the user's username
+    // In the channel header, it seems there is a space after the username, justifying the use of contains.text instead of have.text
+    users.forEach((user) => {
+        cy.get('#channelHeaderTitle').should('be.visible').and('contain.text', user.username);
+    });
+
+    // # Type message and send it to the user
+    cy.get('#post_textbox').
+        type(message).
+        type('{enter}');
+});
+
+/**
+ * Close a DM via the X button
+ * @param {User} sender - TODO
+ * @param {User} recipient - TODO
+ * @param {String} team - TODO
```

Comment:
```suggestion
 * @param {User} sender - the one currently observing and who will close the DM
 * @param {User} recipient - the other user in a DM 
 * @param {String} team - a team where the sender is member of
```

Reply:
Btw, I'm fine to leave this as is though this command is not actually related to this PR. Clean up can be in a separate PR, specifically with `close_current_dm_redirects_spec.js`